kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm-search/src/test/java/org/kie/server/integrationtests/jbpm/search/ProcessSearchServiceIntegrationTest.java

```diff
@@ -0,0 +1,218 @@
+    }
+
+    @Test
+    public void testFindProcessInstanceWithParentIdEqualsToFilter() throws Exception {
+        Map<String, Object> parameters = new HashMap<>();
+        Long processInstanceId  = processClient.startProcess(CONTAINER_ID, PROCESS_ID_EVALUATION, parameters);
+        Assertions.assertThat(processInstanceId).isNotNull();
+        testFindProcessInstanceWithQueryFilter(createQueryFilterEqualsTo(ProcessInstanceField.PARENTPROCESSINSTANCEID, -1), processInstanceId);
+    }
+
+    public void testFindProcessInstanceWithStatusEqualsToFilter() throws Exception {
+        Map<String, Object> parameters = new HashMap<>();
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_EVALUATION, parameters);
+        Assertions.assertThat(processInstanceId).isNotNull();
+        testFindProcessInstanceWithQueryFilter(createQueryFilterEqualsTo(ProcessInstanceField.STATUS, 1), processInstanceId);
+    }
+
+    @Test
+    public void testFindTaskWithAndEqualsToFilter() throws Exception {
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_EVALUATION);
+        Assertions.assertThat(processInstanceId).isNotNull();
+        ProcessInstance process = processClient.getProcessInstance(CONTAINER_ID, processInstanceId);
+        Assertions.assertThat(process).isNotNull();
+
+        HashMap<ProcessInstanceField, Comparable<?>> compareList = new HashMap<>();
+        compareList.put(ProcessInstanceField.PROCESSID, process.getProcessId());
+        compareList.put(ProcessInstanceField.EXTERNALID, CONTAINER_ID);
+        compareList.put(ProcessInstanceField.PROCESSINSTANCEID, processInstanceId);
+        compareList.put(ProcessInstanceField.PROCESSINSTANCEDESCRIPTION, process.getProcessInstanceDescription());
+        compareList.put(ProcessInstanceField.CORRELATIONKEY, process.getCorrelationKey());
+        compareList.put(ProcessInstanceField.USER_IDENTITY, USER_YODA);
+        compareList.put(ProcessInstanceField.PARENTPROCESSINSTANCEID, process.getParentId());
+        compareList.put(ProcessInstanceField.STATUS, process.getState());
+        compareList.put(ProcessInstanceField.PROCESSVERSION, process.getProcessVersion());
+        compareList.put(ProcessInstanceField.PROCESSNAME, process.getProcessName());
+
+        List<Long> resultsIds = new ArrayList<>();
+        List<ProcessInstance> results = null;
+
+        results = searchServicesClient.
+                findProcessInstancesWithFilters(createQueryFilterAndEqualsTo(compareList), 0, 100);
+
+        resultsIds = new ArrayList<>();
+        for (ProcessInstance res : results) {
+            resultsIds.add(res.getId());
+        }
+
+        Assertions.assertThat(results).isNotNull();
+        Assertions.assertThat(results.size()).isGreaterThanOrEqualTo(0);
+        Assertions.assertThat(resultsIds).contains(process.getId());
+
+        final ProcessInstance[] instance = new ProcessInstance[1];
```

Comment:
It is because variables inside lambda expressions should be final or effectively final and I am using it in further down in forEach(). I 'll change it, to make it more readable.